### ISSUE TITLE
[BUG] Object Detection training unavailable for large models after upgrade to 2.13.0

### ISSUE THREAD

**Issue by ale3584** [open-edge-platform/geti_v2#1499]
### 🧩 Issue type

Bug 🐞

### 📄 Description

After upgrading to Intel Geti v2.13.0, issues are observed when working with large Object Detection models:

When clicking Advanced settings, the Data management and Training tabs appear for a few seconds and then disappear.

When attempting to start training, an error occurs:

The server cannot or will not process the current request due to invalid syntax.

This behavior was not present in v2.12.1, where large models functioned normally.

### 💻 Expected behavior

Advanced settings should remain visible and editable.

Training should start successfully, as in v2.12.1.

No server-side “invalid syntax” error should occur.

### 🖼️ Screenshots, video recordings or logs

https://github.com/user-attachments/assets/e90896fd-c4b0-4d28-9beb-a82f475a84bc

### 📦 Geti version

2.13.0

### Hardware

Intel Core i7-14700KF, 128GB RAM, 1.28TB SSD, RTX4080

### 🖥️ Operating System

Ubuntu 22.04

**Comment by piotrgrubicki** [comment]
@ale3584 , could you open a private session in your browser and see if you can still observe this issue? It may be related to the old JS code being cached by your browser.

**Comment by ale3584** [comment]
> [@ale3584](https://github.com/ale3584) , could you open a private session in your browser and see if you can still observe this issue? It may be related to the old JS code being cached by your browser.

I have cleared browser cache in both Chrome and Edge, and also tested in Incognito / Private mode.
The issue persists — the Advanced settings tabs (“Data management”, “Training”) still appear for a few seconds and then disappear.
Starting training for large Object Detection models still results in the error:

The server cannot or will not process the current request due to invalid syntax.

**Comment by piotrgrubicki** [comment]
@ale3584 , most probably, it is the `training_configuration` endpoint that reports a problem.
1. Could you, in your browser, open the developer tools (F12), to go Network, reproduce the problem, and spot the request that return 4xx or 5xx? For the ones with 4xx, could you paste the reponse?
2. Could you attach the log file produced by the following command?
```
sudo kubectl logs $(sudo kubectl get pods -n impt | grep impt-director | awk '{print $1}' | head -n 1) -n impt > director.log
```

**Comment by ale3584** [comment]
1. 400 bad request
URL request
/api/v1/organizations/dfcdb417-bbb3-4225-8271-f2228547259b/workspaces/646331a5-1ece-4d09-a7b1-23106c2f52f2/projects/690c593af9b9926c4106f25f/training_configuration?task_id=690c593af9b9926c4106f262&model_manifest_id=Object_Detection_DFine_X

{
    "error_code": "invalid_payload",
    "errors": [
        {
            "message": "Extra inputs are not permitted",
            "type": "extra_forbidden",
            "location": "hyperparameters.dataset_preparation.augmentation.center_crop"
        },
        {
            "message": "Extra inputs are not permitted",
            "type": "extra_forbidden",
            "location": "hyperparameters.dataset_preparation.augmentation.random_iou_crop"
        }
    ],
    "http_status": 400
}

**Comment by ale3584** [comment]
2. log
[director.log](https://github.com/user-attachments/files/23547338/director.log)

**Comment by ale3584** [comment]
Problem with YOLO-X and DFine-X models. Other models works fine.

**Comment by ale3584** [comment]
There is solution to fix  this manual without reinstall?

**Comment by piotrgrubicki** [comment]
@ale3584 I am checking a workaround, so still cannot be 100% sure if it works. What you could try is: export the dataset from your project (Datumaro format) and use it to create a new project. Then, you should be able to train any model.
(By reinstalling you will loose your data.)

**Comment by ale3584** [comment]
> [@ale3584](https://github.com/ale3584) I am checking a workaround, so still cannot be 100% sure if it works. What you could try is: export the dataset from your project (Datumaro format) and use it to create a new project. Then, you should be able to train any model. (By reinstalling you will loose your data.)

This week we'll finish annotating the dataset. We'll then be able to test whether creating a new project and importing a previously exported dataset will work.

**Comment by piotrgrubicki** [comment]
This problem is a side effect of the change introduced in #1187 
To reproduce:
In 2.12.x, for the models affected by this PR (I used DFine-X), run a training but set a no n default value in Advanced settings.
Then, upgrade to 2.13.0 and you will see the problem described originally here.
Workaround:
Export your dataset to the Datumaro format, and create a new project from this dataset. This way, you will have a clean project configuration and the problem will be gone.

**Comment by ale3584** [comment]
> This problem is a side effect of the change introduced in [#1187](https://github.com/open-edge-platform/geti/pull/1187) To reproduce: In 2.12.x, for the models affected by this PR (I used DFine-X), run a training but set a no n default value in Advanced settings. Then, upgrade to 2.13.0 and you will see the problem described originally here. Workaround: Export your dataset to the Datumaro format, and create a new project from this dataset. This way, you will have a clean project configuration and the problem will be gone.

This helped. But there are pitfalls. I'll describe my case.

I created a project manually, adding 1 label (I have 36 labels in the project). When importing the dataset, the labels weren't added automatically. Then I went to Geti SDK. I uploaded the labels to the project, but when loading, I need to account for the manually created label; it will be considered a duplicate. Remember that the dataset was already loaded, and the labels weren't automatically assigned. I had to assign all the correspondence manually.
Perhaps if I use Geti SDK from creating a new project to importing the dataset, everything will load automatically, but I'll have to check.

**Comment by leoll2** [comment]
> I created a project manually, adding 1 label (I have 36 labels in the project). When importing the dataset, the labels weren't added automatically. Then I went to Geti SDK. I uploaded the labels to the project, but when loading, I need to account for the manually created label; it will be considered a duplicate.

Hi, I can elaborate a bit on the relationship between projects and labels in Geti, I hope it helps to figure out the right sequence of steps for your projects.

Every Geti project is created with two or more labels, specified by the user. It is later possible add/remove labels to an existing project ("Labels" tab in the UI, [`ProjectClient.add_labels`](https://github.com/open-edge-platform/geti-sdk/blob/main/geti_sdk/rest_clients/project_client/project_client.py#L583) in the SDK). Note that models can only predict the labels that exist at training time, therefore if you later add a new label then it's recommended to fine-tune the model again.  
Exported datasets also have labels, namely the categories associated to the annotations in the dataset. When you import a dataset to an existing project, there is a "label mapping" stage where you choose how to map each dataset label to one of the existing project labels. In other words, by design the import operation does NOT create new labels in the project, it only maps labels in the dataset to labels in the project. In your case, you probably want to create all the 36 labels before initiating the dataset import operation.

The above applies to imports to _existing_ projects. Conversely, if you import a dataset to create a _new_ project, then the resulting project will have the same exact labels as the imported dataset.

**Comment by piotrgrubicki** [comment]
If the workaround does not work, pls, provide new details.

**Comment by ale3584** [comment]
> If the workaround does not work, pls, provide new details.

The workaround worked — uploading the dataset into a new project fixed the issue.

**Comment by brian-english1978** [comment]
I have this same issue using Mask Rcnn model with the swin backbone. I am able to use the export work around but my previously working model had been trained on 1000+ epochs and was very robust. Is there no solution besides essentially starting the training process over?

**Comment by piotrgrubicki** [comment]
@brian-english1978 , there is no other solution, I am afraid.

**Comment by brian-english1978** [comment]
Can I run the "update" process but revert to the previous version? Do we know if that would correct the backend web api issues? What's interesting is that the Mask Rcnn model I can no longer initiate training on, is still viable and being used for the auto annotation features when it's chosen as the active model.

**Comment by piotrgrubicki** [comment]
Once upgraded, it is not possible to revert to the previous version.
